docs/hugo/content/contributing/extension-points/arm-resource-modifier.md

```diff
@@ -0,0 +1,243 @@
+---
+title: ARMResourceModifier
+linktitle: ARMResourceModifier
+weight: 10
+---
+
+## Description
+
+`ARMResourceModifier` allows resources to modify the payload that will be sent to Azure Resource Manager (ARM) immediately before it is transmitted. This extension point is invoked after the standard resource conversion and validation, but before the HTTP request is made to ARM.
+
+The interface is invoked during PUT and PATCH operations to ARM, giving the resource an opportunity to make last-minute adjustments to the ARM payload based on runtime conditions, Azure state, or complex business logic that cannot be expressed in the generated code.
+
+## Interface Definition
```

Comment:
Wondering if we should just link to the code here? Or directly embed the code via link (does Hugo support that? probably not).

Calling the exact structure + what the parameters mean out here just seems like writing an interface/function comment and I'd sorta rather have a single place where we document it well (ideally in the code) and then reference that place from the docs?